static/app/components/events/interfaces/frame/utils.tsx

```diff
@@ -149,3 +151,86 @@ export function hasFileExtension(filepath: string) {
   // Check if the filepath matches the pattern
   return fileExtensionPattern.test(filepath);
 }
+
+/**
+ * Extracts the origin URL from an event
+ *
+ * TODO: Should consider other sources of origin besides just url tag
+ *
+ * @param event The event to extract the origin from
+ * @returns The origin URL string, or empty string if not found/invalid
+ */
+function extractEventOrigin(event: Event): string {
+  const urlTag = event.tags.find(({key}) => key === 'url');
+
+  if (!urlTag?.value) {
+    return '';
+  }
+
+  return safeURL(urlTag.value)?.origin ?? '';
+}
+
+/**
+ * Extracts the root domain from a URL string (e.g. "example.com" from "https://sub.example.com/path")
+ *
+ * @param url The URL string to extract the root domain from
+ * @returns The root domain string, or empty string if URL is invalid
+ */
+function getRootDomain(url: string): string {
+  try {
+    const hostname = new URL(url).hostname;
+    // Split hostname into parts and get the last two parts (if they exist)
+    const parts = hostname.split('.');
+    return parts.slice(-2).join('.');
+  } catch {
+    return '';
+  }
+}
+
+/**
+ * Extracts the protocol from a URL string (e.g. "https:" from "https://example.com")
+ *
+ * @param url The URL string to extract the protocol from
+ * @returns The protocol string (including the colon), or empty string if URL is invalid
+ */
+function getProtocol(url: string): string {
+  try {
+    return new URL(url).protocol;
+  } catch {
+    return '';
```

Comment:
May want to catch it and review edge cases. You can later remove logging it to Sentry once you have handled all edge cases.